### ISSUE TITLE
Inject/force-write the css namespace into bundled library

### ISSUE THREAD

**Issue by js-jegan** [styled-components/styled-components#3605]
1. Do you have a usage question?
=============================
Yes

2. Think you found a bug?
=============================
No

3. Do you have a feature request?
=============================
Not sure

## Environment
Prod

## Steps to reproduce
- Create a component using styled-components. say a library.
- Export or Publish the component with `styled-components` as peerDep
- Import the component into your actual project which also has some styled components.
- In order to avoid css collisions, according to this [doc](https://styled-components.com/docs/advanced#avoiding-conflicts-with-thirdparty-styles-and-scripts) add css namespace

## Expected Behavior
Expected the supplied css namespace to be injected for the library components too

## Actual Behavior
supplied css namespace is just been injected to the actual project's styled components.
```
// library - prepare `bundle` & export
export const TestButton = styled.button`
  color: black;
`;

//totally new world - actual project
import { TestButton } from `@some/library-above`

export const NewButton = styled.button`
  color: yellow;
`;

// babel configs
    [
      "@quickbaseoss/babel-plugin-styled-components-css-namespace",
      { "cssNamespace": ".parent-app-el" }
    ],

// Observe in html about the injected styles from styled-components
.xyz { // TestButton style
  color: black;
}

.parent-app-el .abc { // NewButton style
  color: yellow;
}
```

Ways to inject/force-write the css namespaces into bundled library? Yes, using `@quickbaseoss/babel-plugin-styled-components-css-namespace` in library would solve it, but I would like to hand the css namespace responsibility to the consumer of the library.

**Comment by mlewando-cp** [comment]
I have same issue in my project, any luck finding solution for this?

**Comment by js-jegan** [comment]
> I have same issue in my project, any luck finding solution for this?

nope. not yet. Need some love from @styled-components folks. I believe they might have some solution/workaround.

**Comment by quantizor** [comment]
Closing — this issue is stale (2+ years without resolution) and relates to an older version or has been superseded by v6.x changes. If this is still reproducible on the latest v6.x release, please open a new issue with a minimal reproduction.
